be/src/exec/vectorized/sorting/sort_helper.h

```diff
@@ -0,0 +1,211 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#if defined(__SSE2__)
+#include <emmintrin.h>
+#endif
+
+#include "column/nullable_column.h"
+#include "column/type_traits.h"
+#include "column/vectorized_fwd.h"
+#include "exec/vectorized//sorting//sort_permute.h"
+#include "util/orlp/pdqsort.h"
+
+namespace starrocks::vectorized {
+
+// Comparator for sort
+template <class T>
+struct SorterComparator {
+    static int compare(const T& lhs, const T& rhs) {
+        if (lhs == rhs) {
+            return 0;
+        } else if (lhs < rhs) {
+            return -1;
+        } else {
+            return 1;
+        }
+    }
+};
+
+template <>
+struct SorterComparator<Slice> {
+    static int compare(const Slice& lhs, const Slice& rhs) { return lhs.compare(rhs); }
+};
+
+#ifndef NDEBUG
+template <class PermutationType>
+static std::string dubug_column(const Column* column, const PermutationType& permutation) {
+    std::string res;
+    for (auto p : permutation) {
+        res += fmt::format("{:>5}, ", column->debug_item(p.index_in_chunk));
+    }
+    return res;
+}
+#endif
+
+inline static size_t find_byte(const std::vector<uint8_t>& list, size_t start, uint8_t byte) {
+    const uint8_t* ptr = list.data() + start;
+    const uint8_t* end = list.data() + list.size();
+
+#ifdef __SSE2__
+    // use simd to test 16bytes each time
```

Comment:
Does this SIMD instruction have any optimization effect?

Reply:
I wrote a benchmark at `BM_find_zero_simd`, it's 10 times faster then plain version.